mlflow/keras.py

```diff
@@ -449,11 +449,33 @@ def on_train_end(self, logs=None):
             try_mlflow_log(log_model, self.model, artifact_path='model')
 
     def early_stop_check(callbacks):
+        if LooseVersion(keras.__version__) < LooseVersion('2.3.0'):
+            es_callback = keras.callbacks.EarlyStopping
+        else:
+            es_callback = keras.callbacks.callbacks.EarlyStopping
         for callback in callbacks:
-            if isinstance(callback, keras.callbacks.callbacks.EarlyStopping):
+            if isinstance(callback, es_callback):
                 return callback
         return None
 
+    def log_early_stop(callback, history):
+        if callback:
+            stopped_epoch = callback.stopped_epoch
```

Comment:
Since `stopped_epoch` & `patience` aren't publicly-documented fields, we might want to try-catch this block for forwards-compatibility, e.g. define a helper like

```
def _get_callback_attrs(callback):
    try:
        stopped_epoch = callback.stopped_epoch
        patience = callback.patience
        restore_best_weights = callback.restore_best_weights
        return stopped_epoch, patience, restore_best_weights
    except Exception:
        return None
```

and call it here:

```
callback_attrs = _get_epoch_and_patience(callback)
if callback_attrs is None:
  return
epoch, patience, best_weights = callback_attrs
# Proceed as before, using the callback attributes
```